userspace/libsinsp/filter/parser.cpp

```diff
@@ -43,14 +43,73 @@ limitations under the License.
 	#include <regex>
 #endif  // _WIN32
 
+namespace
+{
+enum RGEX_ID
+{
+	RGX_NOTBLANK_ID,
+	RGX_IDENTIFIER_ID,
+	RGX_FIELDNAME_ID,
+	RGX_FIELDARGBARESTR_ID,
+	RGX_HEXNUM_ID,
+	RGX_NUMBER_ID,
+	RGX_BARESTR_ID
+};
+
+const std::string rgex_patterns[] = {
+	"not[[:space:]]+",
+	"[a-zA-Z]+[a-zA-Z0-9_]*",
+	"[a-zA-Z]+[a-zA-Z0-9_]*(\\.[a-zA-Z]+[a-zA-Z0-9_]*)+",
+	"[^][\"'[:space:]]+",
+	"0[xX][0-9a-zA-Z]+",
+	"[+\\-]?[0-9]+[\\.]?[0-9]*([eE][+\\-][0-9]+)?",
+	"[^()\"'[:space:]=,]+",
+};
+
+template<RGEX_ID Rid >
+class rgex_id_class
+{
+private:
+	regex_t m_re;
+public:
+	rgex_id_class()
+	{
+		auto rgx = "^(" + rgex_patterns[Rid] + ")";
+		if(regcomp(&m_re, rgx.c_str(), REG_EXTENDED) != 0)
+		{
+			ASSERT(false);
+		}
+	}
+
+	const regex_t* value() const
+	{
+		return &m_re;
+	}
+
+	~rgex_id_class()
+	{
+		regfree(&m_re);
+	}
+
+	const std::string& rexp() const
+	{
+		return rgex_patterns[Rid];
+	}
+};
+
+};
+
+#define RGEX_ID_CLASS_VAR(ID) rgex_id_class<ID ## _ID> ID
+
+RGEX_ID_CLASS_VAR(RGX_NOTBLANK);
+RGEX_ID_CLASS_VAR(RGX_IDENTIFIER);
+RGEX_ID_CLASS_VAR(RGX_FIELDNAME);
+RGEX_ID_CLASS_VAR(RGX_FIELDARGBARESTR);
+RGEX_ID_CLASS_VAR(RGX_HEXNUM);
+RGEX_ID_CLASS_VAR(RGX_NUMBER);
+RGEX_ID_CLASS_VAR(RGX_BARESTR);
+
 #ifdef USE_POSIX_REGEX
```

Comment:
Connected to the comment above: the whole ifdef should be moved inside `rgex_patterns`.